compiler/luci/pass/src/RemoveUnnecessaryCastPass.cpp

```diff
@@ -25,6 +25,26 @@ namespace
   if (not(cond))                  \
     return false;
 
+/**
+ * BEFORE
+ *
+ *      [CircleNode]
+ *            |
+ *            |
+ *      [CircleCast] (in_data_type == out_data_type)
+ *            |
+ *            |
+ *      [CircleNode]
+ *
+ * AFTER
+ *
+ *      [CircleNode]
+ *            |
+ *            |           [CircleCast removed]
```

Comment:
```suggestion
 *            |           [CircleCast]
```
`CircleCast` is not removed with this pass, it will be removed in dead node elimination pass.

Reply:
Yes, you are right. This was a kind of mental shortcut... :)